component/loki/source/kafka/kafka.go

```diff
@@ -26,6 +27,16 @@ func init() {
 	})
 }
 
+// Takes in a variable of type `rivertypes.Secret`, converts it to an OptionalSecret in order to extract the string value
+func ConvertSecretToString(secret rivertypes.Secret) (string, error) {
```

Comment:
A function like this isn't necessary, you can just cast a `rivertypes.Secret` into a string without any logic:

`string(secretValue)`